tests/host/core/test_string.cpp

```diff
@@ -131,6 +131,20 @@ TEST_CASE("String concantenation", "[core][String]")
     REQUIRE(str == "-100");
     str = String((long)-100, 10);
     REQUIRE(str == "-100");
+    // Non-zero-terminated array concatenation
+    const char buff[] = "abcdefg";
+    String n;
+    n = "1234567890"; // Make it a SSO string, fill with non-0 data
+    n = "1"; // Overwrite [1] with 0, but leave old jump in SSO space still
```

Comment:
junk?

Reply:
Actually, no.  Idea is to fill SSO array[0...10] with non-0 values, to catch if we don't 0-terminate properly in the concat(char*, int) method.  Otherwise if there's a leftover 0 in the SSO array and we don't add one in concat() it may not actually be detected.

Reply:
I did see the typo while fixing the others and fixed it.  D'oh.  I'm still jet lagged.